src/arrayMethodReduce.js

```diff
@@ -3,9 +3,25 @@
 /**
  * Implement method Reduce
  */
+function isEqual(x1, x2) {
+  return x1 === x2 || (Number.isNaN(x1) && Number.isNaN(x2));
+}
+
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let counter = initialValue;
+    let beginOfIteration = 0;
```

Comment:
```suggestion
    let startOfIteration = 0;
```
begin is used as a verb